lib/auth/db.go

```diff
@@ -88,6 +90,17 @@ func (s *Server) GenerateDatabaseCert(ctx context.Context, req *proto.DatabaseCe
 	}, nil
 }
 
+// getCaAndSigner returns correct signer and CA that should be used when generating database certificate.
+// This function covers the database CA rotation scenario when on rotation init phase additional/new TLS
+// key should be used to sign the database CA. Otherwise, the trust chain will break on update_clients.
+func getCaAndSigner(keyStore keystore.KeyStore, databaseCA types.CertAuthority, req *proto.DatabaseCertRequest) ([]byte, crypto.Signer, error) {
```

Comment:
Names like this are always hard to make readable. If you inline it you don't need a name :)

```suggestion
func getCAandSigner(keyStore keystore.KeyStore, databaseCA types.CertAuthority, req *proto.DatabaseCertRequest) ([]byte, crypto.Signer, error) {
```